hw/ip/spi_host/data/spi_host_testplan.hjson

```diff
@@ -66,68 +61,140 @@
+
+            Checking:
+              - verify that all speeds are supported
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_select_timing
+      desc: '''
             Stimulus:
-              - Program ERROR_ENABLE register to enable the TXOVERFLOW interrupt
-              - Program transaction with proper constraints to assert the TXOVERFLOW interrupts
+              - Randomly select a setting for the 16bit clock divider
 
             Checking:
-              - Ensure excess tx data is dropped
-              - Ensure the matching between the value of OVERFLOW bit in the ERROR_STATUS
-                and the ERROR_ENABLE register the once error interrupt pin is asserted
+              - Check that the DUT operates correctly under different SPI clock speeds
             '''
       milestone: V2
-      tests: ["spi_host_tx_overflow_error"]
+      tests: []
     }
     {
       name: component_reset
       desc: '''
             Test components (spi_fsm, rx_fifo, tx_fifo) are randomly reset
 
             Stimulus:
-              - Reset the components of spi_host randomly after a random number of data shows up on fifos
+              - Reset the components of spi_host randomly
+                after a random number of data shows up on fifos
 
             Checking:
               - Ensure that reads to RXDATA register yield 0s after the rx_fifo is reset
               - Ensure that transactions are dropped in both the scoreboard and spi_agent monitor
                 after the tx_fifo or spi_fsm is reset
             '''
       milestone: V2
-      tests: ["spi_host_component_reset"]
+      tests: []
     }
     {
-      name: clock_domain
+      name: modes
       desc: '''
-            TBD - Verify the function of clock-crossing domain for spi_host
+            - Verify the function of spi_host in different special_modes
+              (pass-through and manual_ctrl_cs)
 
             Stimulus:
               - TBD
-
             Checking:
               - TBD
       '''
       milestone: V2
-      tests: ["spi_host_clock_domain"]
+      tests: []
     }
     {
-      name: special_modes
+      name: cpol_cpha
       desc: '''
-            TBD - Verify the function of spi_host in special_modes (passthrough and manual_ctrl_cs)
+            Stimulus:
+              - Randomly chip select for different polarity / phase
 
+            Checking:
+              - Check that the DUT operates correctly under different cs_n settings
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: full_cycle
+      desc: '''
             Stimulus:
-              - TBD
+              - randomly select FULLCYC to be set
 
             Checking:
-              - TBD
-      '''
+              - Check that the data can be read one full cycle after the data was asserted
+            '''
       milestone: V2
-      tests: ["spi_host_special_modes"]
+      tests: []
+    }
+    {
+      name: endian
```

Comment:
This looks like a reasonable test, though I was under the impression that it was not possible to test different parameter values, for reasons very specific to our flow?

I think it would be good to document that this is a test that should be done *in principle* but I imagine that we would get absolutely zero resistance to waiving it, as OpenTitan will never use it.   (It was a feature that was added with an eye to reuse, not to OpenTitan minimum requirements.)  It might be even worthwhile to mark it as low-priority in the testplan, as this is so low priority that the only reason we should put it in the test plan is to document that it is a designed but unexplored feature.

Reply:
The same comments apply to multiple CS lines.  They are not going to be used in the chip, so we should perhaps consider documenting them as features we have no immediate plans of testing.

Reply:
@martin-lueker I thought about that - but I am unsure about how to capture "planned but waived" as I think it is important information.
@weicaiyang @sriyerg any comments on how to track deferred testscases?


for multiple cs_lines I think that as long as it is listed as a supported feature I think we should test it.

Reply:
Agree with Rasmus, it seems weird that our spec says we support a feature, but it's not verified. In my previous experience, if some features aren't verified, we will document those aren't supported and user can use those with their own risk.

Reply:
Okay, well, perhaps I can just enumerate them as implemented, but not supported/not verified.

Reply:
Documentation to be changed as part of #7553

Reply:
If the design is parameterized (i.e. it does not refer to the parameters from some `foo_pkg`), then we need to parameterize the UVM testbench as well, so that its easy to get other 'flavors' verified later when they get used in a different OT implementation.